watcher/tool/watcher/rollups.go

```diff
@@ -75,17 +75,26 @@ func receiveAndManageRollups(ctx context.Context, client *influxdb.Client, ch <-
 				for _, rollup := range rollups {
 					switch update.EventType {
 					case watch.Added:
-						err := client.CreateRollup(rollup)
+						err := retry(ctx, func() error {
+							err := client.CreateRollup(rollup)
+							return err
+						})
 						if err != nil {
 							log.Errorf("failed to create rollup %v: %v", rollup, trace.DebugReport(err))
 						}
 					case watch.Deleted:
-						err := client.DeleteRollup(rollup)
+						err := retry(ctx, func() error {
+							err := client.DeleteRollup(rollup)
+							return err
+						})
 						if err != nil {
 							log.Errorf("failed to delete rollup %v: %v", rollup, trace.DebugReport(err))
 						}
 					case watch.Modified:
-						err := client.UpdateRollup(rollup)
+						err := retry(ctx, func() error {
```

Comment:
A couple of concerns:
  - there may be multiple changes queued for the configuration map. Running blocking retry will prevent other changes from taking effect.
  - I think you'd need to implement circuit breakers to break out of retry if the operation is never succeeding (i.e. the error is such that you cannot successfully complete the API). This is API-specific - i.e. for `CreateRollup` that would be checking whether the error is not from failing `Check` (or ensure the preconditions prior to invoking `CreateRollup`).

I would probably have another goroutine to handle updates which you could structure as a loop over a list of handlers - as soon as the handler has succeeded it is removed from the list:
```go
func retryLoop(ctx context.Context, retryC <-chan func() error) {
	var handlers []func() error
	timer := time.NewTimer(5 * time.Second)
	defer timer.Stop()
	for {
		select {
		case handler := <-retryC:
			handlers = append(handlers)
		case <-timer.C:
			for i, handler := range handlers {
				if err := handler(); err != nil {
					log.WithError(err).Warn("Failed to complete handler.")
					continue
				}
				// Remove handler
				handlers = append(handlers[:i], handlers[i+1]...)
			}
		case <-ctx.Done():
			return
		}
	}
}
func receiveAndManageRollups(ctx context.Context, client *influxdb.Client, ch <-chan kubernetes.ConfigMapUpdate, retryC chan<- func() error) {
	// ...
	for _, rollup := range rollups {
		switch update.EventType {
		case watch.Added:
			handler := func() error {
				err := client.CreateRollup(rollup)
				return err
			}
			if err := handler(); err == nil {
				break
			}
			log.WithError(err).Warnf("Failed to create rollup %v", rollup)
			select {
			case retryC <- handler:
				// Queue handler on retry list
			case <-ctx.Done():
			}
		// ...
}
```